### PR TITLE
Add link to documentation when build fails due to ambigious name expa…

### PR DESCRIPTION
**Pull request by donat**
Fixes
- #24395

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this please

File: subprojects/core/src/main/java/org/gradle/execution/TaskSelectionException.java [hunk 1]
```diff
@@ -36,5 +37,8 @@ public void appendResolutions(Context context) {
             context.getClientMetaData().describeCommand(output.withStyle(UserInput), ProjectInternal.TASKS_TASK);
             output.text(" to get a list of available tasks.");
         });
+        context.appendResolution(output -> {
```
[the remark this answers is not in the document]
Comment (donat) [reply subprojects/core/src/main/java/org/gradle/execution/TaskSelectionException.java]: Thanks for spotting that. I've extracted the suggestion to a common base class for consistency.

[the remark this answers is not in the document]
Comment (reinsch82) [reply subprojects/core/src/main/java/org/gradle/execution/TaskSelectionException.java]: even better :)

**Review by reinsch82 — APPROVED** [review]
LGTM

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**